client/src/partials/reports/global_transaction/global_transaction.html

```diff
@@ -61,8 +64,11 @@
 
             <div class="panel-footer">
               <div class="pull-right">
-                <button class="btn btn-sm btn-success" type="button" ng-click="generate()" ng-disabled="configForm.$invalid">
-                  {{ "UTIL.GENERATE" | translate }}
+                <button class="btn btn-sm btn-success" type="button" ng-click="ReportCtrl.generate()" ng-disabled="configForm.$invalid">
+                  <span ng-if="!ReportCtrl.session.buttonLoader">{{ "UTIL.GENERATE" | translate }}</span>
+                  <span ng-if="ReportCtrl.session.loaderState === 'loading' && ReportCtrl.session.buttonLoader">
+                    <loading-indicator></loading-indicator>
```

Comment:
That would be nice, unfortunately, AngularJS recommends _not_ creating "self closing" directives.  Browsers sometimes have trouble parsing them.  [Here's](https://github.com/angular/angular.js/issues/1237) an AngularJS issue about this problem.  [Here](https://github.com/angular/angular.js/issues/1953) is another one.